File: src/design_patterns/behavioral/strategy/c#/strategy.cs
interface IStrategy
{
    int DoOperation(int num1, int num2);
}

class OperationAdd : IStrategy
{
    public int DoOperation(int num1, int num2)
    {
        return num1 + num2;
    }
}

class OperationSubtract : IStrategy
{
    public int DoOperation(int num1, int num2)
    {
        return num1 - num2;
    }
}

class OperationMultiply : IStrategy
{
    public int DoOperation(int num1, int num2)
    {
        return num1 * num2;
    }
}

class Context
{
    private IStrategy strategy;

    public Context(IStrategy strategy)
    {
        this.strategy = strategy;
    }

    public int ExecuteStrategy(int num1, int num2)
    {
        return strategy.DoOperation(num1, num2);
    }
}

File: src/design_patterns/behavioral/strategy/python/strategy.py
class Strategy:
    """ """
    def do_operation(self, num1, num2):
        """

        :param num1: param num2:
        :param num2: 

        """
        pass


class OperationAdd(Strategy):
    """ """
    def do_operation(self, num1, num2):
        """

        :param num1: param num2:
        :param num2: 

        """
        return num1 + num2


class OperationSubtract(Strategy):
    """ """
    def do_operation(self, num1, num2):
        """

        :param num1: param num2:
        :param num2: 

        """
        return num1 - num2


class OperationMultiply(Strategy):
    """ """
    def do_operation(self, num1, num2):
        """

        :param num1: param num2:
        :param num2: 

        """
        return num1 * num2


class Context:
    """ """
    def __init__(self, strategy):
        self.strategy = strategy

    def execute_strategy(self, num1, num2):
        """

        :param num1: param num2:
        :param num2: 

        """
        return self.strategy.do_operation(num1, num2)
